Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelTask.kt

```diff
@@ -0,0 +1,166 @@
+            // risk level
+            if (!isNetworkEnabled(context)) {
+                RiskLevelRepository.setLastCalculatedRiskLevelAsCurrent()
+                return Result(UNDETERMINED)
+            }
+            with(riskLevels) {
+                return Result(
+                    when {
+                        !enfClient.isTracingEnabled.first().also {
+                            checkCancel()
+                        } -> NO_CALCULATION_POSSIBLE_TRACING_OFF
+
+                        calculationNotPossibleBecauseOfNoKeys().also {
+                            checkCancel()
+                        } -> UNKNOWN_RISK_INITIAL
+
+                        calculationNotPossibleBecauseOfOutdatedResults().also {
+                            checkCancel()
+                        } -> if (backgroundJobsEnabled)
+                            UNKNOWN_RISK_OUTDATED_RESULTS
+                        else
+                            UNKNOWN_RISK_OUTDATED_RESULTS_MANUAL
+
+                        isIncreasedRisk(getNewExposureSummary()).also {
+                            checkCancel()
+                        } -> INCREASED_RISK
+
+                        !isActiveTracingTimeAboveThreshold().also {
+                            checkCancel()
+                        } -> UNKNOWN_RISK_INITIAL
+
+                        else -> LOW_LEVEL_RISK
+                    }.also {
+                        checkCancel()
+                        updateRepository(it, System.currentTimeMillis())
+                    }
+                )
+            }
+        } catch (error: Exception) {
+            Timber.tag(TAG).e(error)
+            error.report(ExceptionCategory.EXPOSURENOTIFICATION)
+            throw error
+        } finally {
+            Timber.i("Finished (isCanceled=$isCanceled).")
+            internalProgress.close()
+        }
+    }
+
+    /**
+     * If there is no persisted exposure summary we try to get a new one with the last persisted
+     * Google API token that was used in the [de.rki.coronawarnapp.transaction.RetrieveDiagnosisKeysTransaction]
+     *
+     * @return a exposure summary from the Google Exposure Notification API
+     */
+    private suspend fun getNewExposureSummary(): ExposureSummary {
+        val googleToken = LocalData.googleApiToken()
+            ?: throw RiskLevelCalculationException(IllegalStateException("Exposure summary is not persisted"))
+
+        val exposureSummary =
+            InternalExposureNotificationClient.asyncGetExposureSummary(googleToken)
+
+        return exposureSummary.also {
+            Timber.tag(TAG)
+                .v("Generated new exposure summary with $googleToken")
+        }
+    }
+
+    private fun checkCancel() {
+        if (isCanceled) throw TaskCancellationException()
+    }
+
+    private val backgroundJobsEnabled: Boolean
+        get() = ConnectivityHelper.autoModeEnabled(CoronaWarnApplication.getAppContext()).also {
```

Comment:
We have a non static variant available for that, see `BackgroundModeStatus`, Could we use that here?